bin/echo.c

```diff
@@ -71,6 +72,8 @@ static int wait_for_event(int fd, s2n_blocked_status blocked)
 
 int negotiate(struct s2n_connection *conn, int fd)
 {
+    POSIX_ENSURE_REF(conn);
+
```

Comment:
Is this necessary for your change, and does it use any private symbols? (How is POSIX_BAIL defined?)

Reply:
As discussed offline: We currently use a number of internal safety macros defined within https://github.com/aws/s2n-tls/blob/main/utils/s2n_safety_macros.h  in here which doesn't seem correct. I can create an issue for it. We should prevent usage of internal safety macros within bin/s2nd and s2nc.